packages/echo-core/src/services/analytics/analyticsTelemetryFilter.test.ts

```diff
@@ -0,0 +1,54 @@
+import { ITelemetryItem } from '@microsoft/applicationinsights-core-js';
+import { analyticsConfiguration } from './analytics';
+import { telemetryFilterShouldInclude } from './analyticsTelemetryFilter';
+import { AnalyticsNameFilterFunction } from './analyticsTypes';
+
+const telemetryItem: ITelemetryItem = { name: 'name', baseData: { name: 'baseDataName' } };
+
+describe('telemetryFilterShouldInclude & addTelemetryNameFilter', () => {
+    it(`no filters should not exclude`, () => {
+        const actual = telemetryFilterShouldInclude(telemetryItem);
+        expect(actual).toEqual(true);
+    });
+
+    it(`after add filter it should exclude, and after remove filter it shouldn't exclude`, () => {
+        const telemetryExcludeFunctionMatch: AnalyticsNameFilterFunction = {
+            shouldExclude: (name: string): boolean => name === 'baseDataName'
+        };
+        analyticsConfiguration.addTelemetryNameFilter(telemetryExcludeFunctionMatch);
+        const actualExcluded = telemetryFilterShouldInclude(telemetryItem);
+        expect(actualExcluded).toEqual(false);
+
+        analyticsConfiguration.removeTelemetryNameFilter(telemetryExcludeFunctionMatch);
+        const actualIncluded = telemetryFilterShouldInclude(telemetryItem);
+        expect(actualIncluded).toEqual(true);
+    });
+
+    it(`should not exclude if filter has mismatch on baseData.name`, () => {
+        const telemetryExcludeFunctionMisMatch: AnalyticsNameFilterFunction = {
+            shouldExclude: (name: string): boolean => name === 'mismatchOnName'
+        };
+        analyticsConfiguration.addTelemetryNameFilter(telemetryExcludeFunctionMisMatch);
+        const actualIncluded = telemetryFilterShouldInclude(telemetryItem);
+        expect(actualIncluded).toEqual(true);
+
+        analyticsConfiguration.removeTelemetryNameFilter(telemetryExcludeFunctionMisMatch);
+    });
+
+    it(`multiple filters should exclude if 1 filter is matching baseData.name`, () => {
+        const telemetryExcludeFunctionMisMatch: AnalyticsNameFilterFunction = {
```

Comment:
We didn't agree on how we should write tests yet:
But I would suggest to mark the setup, execution and assertion phases somehow, just to make the test case more readable.

Like in here forexample:
`````
        it('should return with a default errorMessage object', () => {
            // given
            const backendError = new BackendError({ ...exceptionDetails });
            const expectedMessage = {
                friendlyMessage:
                    'There was a server error. Please try again later. If the issue persists, contact the Echo InField team with the following id: frontEnd_4fzzzxjyl.',
                errorType: 'BackendError'
            };

            // when
            const actual = getUserFriendlyErrorMessage(backendError);

            // then
            expect(actual).toEqual(expectedMessage);
        });
`````